IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework.Tests/TestBuilderTests/PositiveConditions/LoggerTests.cs

```diff
@@ -0,0 +1,188 @@
+                .AddTestBlock<RemovedLogBlockProp>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void RemovedLoggerDoesNotThrowWhenAttemptingToActivateCtor()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .RemoveLogger()
+                .AddTestBlock<RemovedLogBlockCtor>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void RemovedLoggerDoesNotThrowWhenAttemptingToActivateExecuteArg()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .RemoveLogger()
+                .AddTestBlock<RemovedLogBlockExecuteArg>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void CustomLoggerAddsWithoutError()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .AddLogger<CustomLogger>()
+                .AddTestBlock<CustomLogBlock>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void RemovingLoggerTwiceDoesNotThrow()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .RemoveLogger()
+                .RemoveLogger()
+                .AddTestBlock<RemovedLogBlockProp>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+
+        [Fact]
+        public void AddingLoggerThanRemovingDoesNotThrow()
+        {
+            // Arrange
+            TestCase tc = new TestBuilder()
+                .AddLogger<CustomLogger>()
+                .RemoveLogger()
+                .AddTestBlock<RemovedLogBlockProp>()
+                .Build();
+
+            // Act / Assert
+            tc.Execute();
+        }
+    }
+
+    public class DefaultLogBlock : TestBlock
+    {
+        public void Execute()
+        {
+            Assert.NotNull(Log);
+            Assert.IsType<DebugLogger>(Log);
+        }
+    }
+
+    public class RemovedLogBlockProp : TestBlock
+    {
+        public void Execute()
+        {
+            Assert.Null(Log);
+        }
+    }
+
+    public class RemovedLogBlockCtor : TestBlock
+    {
+        public RemovedLogBlockCtor(ITestCaseLogger? log)
+        {
+            Log = log;
+        }
+
+        public void Execute()
+        {
+            Assert.Null(Log);
+        }
+    }
+
+    public class RemovedLogBlockExecuteArg : TestBlock
+    {
+        public void Execute(ITestCaseLogger? log)
+        {
+            Assert.Null(log);
+        }
+    }
+
+    public class CustomLogBlock : TestBlock
+    {
+        public void Execute()
+        {
+            Assert.NotNull(Log);
+            Assert.IsType<CustomLogger>(Log);
+            if(Log is CustomLogger cl)
```

Comment:
I would direct cast since you know the cast will succeed.

```suggestion
            CustomLogger cl = (CutomLogger)Log;
```